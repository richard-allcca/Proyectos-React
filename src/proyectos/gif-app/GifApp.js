import React, { useState } from "react";
import FormSearch from "./git-components/FormSearch";
import ListCategories from "./git-components/ListCategories";

const GifApp = () => {
  const [categories, setCategories] = useState("");

  const handleAdd = (inputValue) => {
    setCategories(inputValue);
  };

  return (
    <div className="gif-app-container">
      <h2>Soy una App Gif</h2>
      <FormSearch handleAdd={handleAdd} />

      <hr />
      {categories && (
        <ListCategories
          key={categories}
          category={categories}
          url={`http://api.giphy.com/v1/gifs/search?q=${categories}&limit=10&api_key=iGk4Cf4Uc0afvcm6bNLr15qT3COxulwj`}
        />
      )}
    </div>
  );
};

export default GifApp;
