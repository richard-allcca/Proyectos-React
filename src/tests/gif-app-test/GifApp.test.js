import { shallow } from "enzyme";
import React from "react";
import GifApp from "./../../../../curso-react-resumen/src/proyectos/gif-app/GifApp";

describe("Prueba sobre el <GifApp/>", () => {
  let wrapper = shallow(<GifApp />);

  beforeEach(() => {
    wrapper = shallow(<GifApp />);
  });

  test("Should mostrar el contenido del componente <GifApp/>", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
