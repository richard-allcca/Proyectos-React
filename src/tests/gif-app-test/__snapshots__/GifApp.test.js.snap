// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre el <GifApp/> Should mostrar el contenido del componente <GifApp/> 1`] = `
<div
  className="gif-app-container"
>
  <h2>
    Soy una App Gif
  </h2>
  <FormSearch
    handleAdd={[Function]}
  />
  <hr />
</div>
`;
